presto-jdbc/src/main/java/io/prestosql/jdbc/ConnectionProperties.java

```diff
@@ -128,6 +137,49 @@ public Password()
         }
     }
 
+    private static class Roles
+            extends AbstractConnectionProperty<Map<String, ClientSelectedRole>>
+    {
+        public Roles()
+        {
+            super("roles", NOT_REQUIRED, ALLOWED, Roles::parseRoles);
+        }
+
+        // Roles consists of a list of catalog role pairs.
+        // E.g., `jdbc:presto://example.net:8080/?roles=catalog1:none;catalog2:all;catalog3:role` will set roles `none` in `catalog1`, `all` in `catalog2` and `role` in `catalog3`
+        public static Map<String, ClientSelectedRole> parseRoles(String roles)
+        {
+            return MAP_ENTRIES_SPLITTER.splitToList(roles).stream()
+                    .map(Roles::parseRole)
+                    .collect(toImmutableMap(entry -> entry.get(0), entry -> mapToClientSelectedRole(entry.get(1))));
+        }
+
+        private static ClientSelectedRole mapToClientSelectedRole(String role)
+        {
+            if (ALL.name().equalsIgnoreCase(role)) {
+                return new ClientSelectedRole(ALL, Optional.empty());
+            }
+            if (NONE.name().equalsIgnoreCase(role)) {
+                return new ClientSelectedRole(NONE, Optional.empty());
+            }
+            return new ClientSelectedRole(ClientSelectedRole.Type.ROLE, Optional.of(role));
```

Comment:
What if the role has a name "all"?
In SQL, `SET ROLE "all"` is different than from `SET ROLE ALL`.

(For now it's sufficient to validate role does not contain `"` so that we are future-proof)